data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -429,6 +429,45 @@ class ApiServerClient {
       });
   }
 
+  /**
+   * @description Get Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Unique id of resource
+   * @param {string} opts.resourceType - Name of operation
+   * @param {string} opts.stateList - State of resorce
+   */
+  getResourceListByState(opts) {
+    logger.debug('Get resource with opts: ', opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to get resource list`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to get resource list`);
+    assert.ok(opts.stateList, `Property 'stateList' is required to fetch resource list`);
+    return Promise.try(() => this.init())
+      .then(() => apiserver.apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[opts.resourceType].get({
+          qs: {
+            labelSelector: `state in (${_.join(opts.stateList, ',')})`
+          }
+        }))
+      .then(resources => _.get(resources.body, 'items'))
+      .map(resource => {
+        _.forEach(resource.spec, (val, key) => {
+          try {
+            resource.spec[key] = JSON.parse(val);
+          } catch (err) {
+            resource.spec[key] = val;
+          }
+        });
+        _.forEach(resource.status, (val, key) => {
```

Comment:
Would it be possible for you to iterate asynchronously here to avoid blocking the event loop?

Reply:
https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/435#discussion_r225513229